runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -328,8 +328,8 @@ public static StreamingDataflowWorker fromOptions(DataflowWorkerHarnessOptions o
         threadName ->
             Executors.newSingleThreadScheduledExecutor(
                 new ThreadFactoryBuilder().setNameFormat(threadName).build());
-    GrpcWindmillStreamFactory windmillStreamFactory =
-        createWindmillStreamFactory(options, clientId);
+    GrpcWindmillStreamFactory.Builder windmillStreamFactoryBuilder =
+        createWindmillStreamFactoryBuilder(options, clientId);
     GrpcDispatcherClient dispatcherClient = GrpcDispatcherClient.create(createStubFactory(options));
 
     // If ComputationConfig.Fetcher is the Streaming Appliance implementation, WindmillServerStub
```

Comment:
There is some kind of a cyclic dependency between windmillStreamFactory and computationStateCache and the creation order is different between appliance and SE, which makes the code hard to reason about. How about something like https://github.com/arunpandianp/beam/commit/d428be00f38895a4a9d4ff830619aabe66134796? Then we don't need these comments explaining if something then it is appliance

Reply:
done